File: ios/RoutineChart/RoutineChart/Core/Domain/Models/AuthUser.swift
//
//  AuthUser.swift
//  RoutineChart
//
//  Created for Phase 2.1: Firebase Auth
//

import Foundation

/// Represents an authenticated user (separate from domain User)
struct AuthUser: Identifiable {
    let id: String  // Firebase UID
    let email: String?
    let isAnonymous: Bool
    
    init(id: String, email: String?, isAnonymous: Bool = false) {
        self.id = id
        self.email = email
        self.isAnonymous = isAnonymous
    }
}

File: android/app/src/main/java/com/HammersTech/RoutineChart/core/domain/models/AuthUser.kt
package com.HammersTech.RoutineChart.core.domain.models

/**
 * Represents an authenticated user (separate from domain User)
 * Phase 2.1: Firebase Auth
 */
data class AuthUser(
    val id: String, // Firebase UID
    val email: String?,
    val isAnonymous: Boolean = false,
)
